storage/innobase/xtrabackup/src/ddl_tracker.cc

```diff
@@ -288,6 +288,36 @@ typedef struct {
   space_id_to_name_t *new_tables;
 } copy_thread_ctxt_t;
 
+/**
+ * This function extracts the database name and space name from the
+ * tablepsace path. If the space is not file-per-table tablespace, it
+ * returns only the space name with appended extension. For other cases,
+ * it constructs a path in the format "database_name/space_name"
+ * with an appended extension.
+ *
+ * @param node File node of a tablespace
+ * @return A string representing the destination name
+ */
+std::string get_dest_name(fil_node_t *node) {
+  std::string path{node->name};
+  auto last_separator_pos = path.find_last_of(Fil_path::SEPARATOR);
+
+  std::string db_name = path.substr(0, last_separator_pos);
+  std::string space_name = path.substr(last_separator_pos + 1);
+
+  if (!fsp_is_file_per_table(node->space->id, node->space->flags)) {
+    return space_name + EXT_NEW;
+  }
+
+  // Extract the last part of db_name after the last separator
+  last_separator_pos = db_name.find_last_of(Fil_path::SEPARATOR);
+  if (last_separator_pos != std::string::npos) {
+    db_name = db_name.substr(last_separator_pos + 1);
+  }
+
```

Comment:
also the same for db_name.

add debug assertion here that db_name shouldn't be empty/(not size 0)